### PR TITLE
Code Insights: Add integration test for dashboard crud operations

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/34001

In this PR we cover the dashboard test with integration tests. We didn't use unit test because our units at the moment are strictly bound to concrete implementation and it might be a problem in future when we will have a big shift in GQL/API paradigm in the near future. I wish we had more unit tests tooling around code insights logic but unfortunately, we don't have it at the moment. 
 
## Test plan
- Make sure that CI is green and the integration tests work properly

## App preview:

- [Web](https://sg-web-vk-add-integration-test-for.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-yhalyifrrs.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-menu/DashboardMenu.tsx [hunk 1]
```diff
@@ -50,12 +51,13 @@ export const DashboardMenu: React.FunctionComponent<React.PropsWithChildren<Dash
                 {menuPermissions.configure.display && (
                     <MenuItem
                         as={Button}
+                        outline={true}
                         disabled={menuPermissions.configure.disabled}
                         data-tooltip={menuPermissions.configure.tooltip}
                         data-placement="right"
                         className={styles.menuItem}
+                        data-testid="ConfigureDashboardButton"
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-menu/DashboardMenu.tsx:59]: Nitpick: It bothers me that we're mixing cases in the same file for these test ids. WDYT about choosing camel or kebab case for both of these?

**Review by unclejustin — APPROVED** [review]
Nothing blocking but I do have one general comment. Everywhere you added data-testids I'm wondering if we don't have a better way to select them using accessible standards. I linked a blog post about using ARIA-based items for selection that looks interesting.

This way we can remove all the extra test ids and we will get a11y enforcement for free.

Relatedly I'd like to see something after the expectations that assert deep equal on the variables that check for user-facing changes on the page. i.e. "dashboard saved" or whatever.

I'm also wondering if checking these variables is too much implementation detail. Sure we're asserting the API contract hasn't been broken, but should the UI actually care? Wouldn't we be better served just checking for UI changes and asserting the API contract in unit tests elsewhere?

File: client/web/src/integration/insights/insight-dashboard.test.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                            },
+                        },
+                    },
+                }),
+            },
+        })
+
+        await driver.page.goto(driver.sourcegraphBaseUrl + '/insights/dashboards/all')
+
+        await driver.page.waitForSelector('[data-testid="AddDashboardButton"]')
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/integration/insights/insight-dashboard.test.ts:167]: I think you're right. In fact, we already use the aria-labels selector somewhere in our integration test selectors. I will redo selectors with aria selectors in mind, thanks!

**Comment by vovakulikov** [comment]
> Wouldn't we be better served just checking for UI changes and asserting the API contract in unit tests elsewhere? 

@unclejustin I agree with this statement. The problem that stops us from doing this IMO is that in this case, we should have a kind of working version of our BE but for integration tests or we should run real e2e tests. This is because right after mutation the next thing to do is to update the existing query that this mutation affects.

Like the creation of a new dashboard, triggers get dashboard query and therefore we see a newly created dashboard in the dashboard select. In the case of integration tests, we somehow should catch the mutation and updates our local integration BE store of code insights dashboards. And we don't have anything to do it now, and I'm not sure actually that this is the way to do a proper integration test.

**Pull request merged**